haskell/private/haskell_impl.bzl

```diff
@@ -103,6 +105,10 @@ use the 'haskell_import' rule instead.
                 depset([c.objects_dir]),
             ]),
             compiler_flags = ctx.attr.compiler_flags,
+            # NOTE We can't have profiling and dynamic code at the
+            # same time, see:
+            # https://ghc.haskell.org/trac/ghc/ticket/15394
+            dynamic = False,
```

Comment:
Until that Trac issue is resolved, do we think it's worth having the rule print out a warning when `dynamic and with_profiling`?

Reply:
I considered this. But my understanding of Bazel philosophy is that all warnings should either be an error or are useless (when you're working with large code bases you'll just be drowned in warnings making them not actionable). So the choice is between making this an error and silently proceeding. I at first made this an error, but this project's own test suite quickly made it clear that making it an error was going to be very annoying. I dunno. Maybe we should make this an error? We could use tags to filter out tests that are known to fail when compiling in debug mode.

Reply:
So I went with a warning in the end, as you suggested. We'll see how it goes.